lib/cloudregister/registerutils.py

```diff
@@ -2042,3 +2279,118 @@ def __mv_file_backup(filename):
     message = 'File not preserved' if failed else 'File preserved'
     logging.info(message)
     return failed
+
+
+# ----------------------------------------------------------------------------
+def __set_registries_conf_podman(private_registry_fqdn):
+    """Set the registry search order for Podman."""
+    registries_conf = {}
+    if os.path.exists(REGISTRIES_CONF_PATH):
+        registries_conf, failed = get_registry_conf_file(
+            REGISTRIES_CONF_PATH, 'podman'
+        )
+        if failed:
+            return False
+
+    public_registry_fqdn = 'registry.suse.com'
+    unqualified_search_reg = []
+    unqualified_search_reg = registries_conf.get(
+        'unqualified-search-registries', []
+    )
+    modified = False
+    if unqualified_search_reg:
+        priv_index = -1
+        pub_index = -1
+        if private_registry_fqdn in unqualified_search_reg:
+            priv_index = unqualified_search_reg.index(private_registry_fqdn)
+        if public_registry_fqdn in unqualified_search_reg:
+            pub_index = unqualified_search_reg.index(public_registry_fqdn)
+
+        if not priv_index == 0 or not pub_index == 1:
+            if priv_index > 0:
+                unqualified_search_reg.pop(priv_index)
+            if pub_index > 0:
+                unqualified_search_reg.pop(pub_index)
+            modified = True
+
+    if modified or not unqualified_search_reg:
+        [
+            unqualified_search_reg.insert(0, fqdn) for fqdn in
+            [public_registry_fqdn, private_registry_fqdn]
+        ]
+
+    private_registry = {'location': private_registry_fqdn, 'insecure': False}
+    # a list of dictionaries
+    registry_mirrors = registries_conf.get('registry', [])
+    present = False
+    if private_registry in registry_mirrors:
+        # exact match
```

Comment:
matching with ```in``` is a substring match not an exact match as the comment indicates, is this correct ?

Reply:
but `registry_mirrors` is a list of dictionaries ?
so that `if` checks the following

`{'location': registry-ec2.susecloud.net', 'insecure': False} in [{'location': 'foo'}, {'location': registry-ec2.susecloud.net', 'insecure': False}]`

Reply:
ok for this datatype match it's exact and correct. All good sorry for the noise